### ISSUE TITLE
Integrating Custom VSO Custom Task with GitVersion

### ISSUE THREAD

**Issue by johnkattenhorn** [GitTools/GitVersion#586]
Not an issue just to let you know that if anyone is interested I've started to but together a custom build task that should be able to use the output of the GitVersion custom task to package and push an Octopus Deploy package.

The work done here the team helped me understand custom TFS task so thanks!

You can find my WIP here https://github.com/johnkattenhorn/OctoTFS, should be able to test and complete tomorrow.

**Comment by JakeGinnivan** [comment]
If it uses Octopack under the hood there is also https://github.com/OctopusDeploy/OctoPack/pull/69 in progress which will allow reading GitVersion metadata out of the project.

The OctoTFS task shouldn't be tightly coupled to the GitVersion task if at all possible. Maybe you could have a field which lets you specify the version number and in that you could simply put $(GitVersion_NuGetVersion) or something? That way they stay loosely coupled.

Thanks for letting us know, I'm going to close this issue but I still will get notifications so we can continue the discussion

**Comment by johnkattenhorn** [comment]
Thanks @JakeGinnivan, I saw that open issue too and I agree on the version point, I've just got a Custom Task variable that I was hoping to pass in the GitVersion variable into.

I reached out to Damien at Octopus on their support channel and he given me some tips. In meetings this morning but I'll try to get something working this afternoon.

It's pretty close, just trying to figure out how best to test it without having to deploy endless versions of the script until I get it right given my PowerShell knowledge :-)
